ruby/bps-kafka/lib/bps/kafka.rb

```diff
@@ -4,34 +4,52 @@
 module BPS
   module Kafka
     # ReliablePublisher is a slow, but reliable Kafka Publisher.
-    # There is no batching at all, messages are emitted synchronously.
+    # There is no batching at all, messages are delivered one by one.
     class ReliablePublisher < BPS::Publisher::Abstract
-      DELIVER_MESSAGE_RETRIES = 3 # 1 by kafka-ruby's Kafka::Client default, increase slightly
-
       class Topic < BPS::Publisher::Topic::Abstract
-        def initialize(client, topic_name)
-          @client = client
-          @topic_name = topic_name
+        def initialize(producer, topic_name, opts = {})
+          @producer = producer
+          @opts = opts.merge(topic: topic_name)
         end
 
         def publish(msg_data)
-          @client.deliver_message(msg_data, topic: @topic_name, retries: DELIVER_MESSAGE_RETRIES)
+          @producer.produce(msg_data, **@opts)
+          @producer.deliver_messages
           nil
         end
       end
+      private_constant :Topic
+
+      def initialize(seed_brokers, opts = {})
+        @producer_opts = extract_prefixed_opts!(opts, 'producer_')
```

Comment:
another example of overthinking. just accept opts as they are pls. we can attach additional options on the actual `produce(msg, **opts)` call

Reply:
@dim it's hard to tell apart opts for client/producer then. Prefixing is much easier than enumerating each option key for client/producer.

Reply:
pls don't ... please!!